### PR TITLE
Add from_timestamp_nanos

### PR DESCRIPTION
**Pull request by Ali-Mirghasemi**
Add and implement `from_timestamp_nanos` and add unit test for it

### Known issue

if you input `i64::MAX` function give you a valid datetime, but for `i64::MIN` it gives you `None`
i think it's because time unit is nanoseconds

### Thanks for contributing to chrono!

If your feature is semver-compatible, please target the 0.4.x branch;
the main branch will be used for 0.5.0 development going forward.

Please consider adding a test to ensure your bug fix/feature will not break in the future.

### CODE REVIEW

File: src/naive/datetime/mod.rs [hunk 1]
```diff
@@ -197,6 +197,39 @@ impl NaiveDateTime {
...
+    /// // Negative timestamps (before the UNIX epoch) are supported as well.
+    /// let timestamp_nanos: i64 = -2208936075_000_000_000; //Mon Jan 01 1900 14:38:45 GMT+0000
+    /// let naive_datetime = NaiveDateTime::from_timestamp_nanos(timestamp_nanos);
+    /// assert!(naive_datetime.is_some());
+    /// assert_eq!(timestamp_nanos, naive_datetime.unwrap().timestamp_nanos_opt().unwrap());
+    /// ```
+    #[inline]
+    #[must_use]
+    pub const fn from_timestamp_nanos(nanos: i64) -> Option<NaiveDateTime> {
+        let secs = nanos.div_euclid(1_000_000_000);
```
Comment (djc) [inline src/naive/datetime/mod.rs:227]: This should probably reuse the `NANOS_PER_SEC` `const` that we have defined somewhere.

Comment (Ali-Mirghasemi) [reply src/naive/datetime/mod.rs:227]: `NANOS_PER_SEC` is in duration.rs mod and is inaccessible for me to use it, the solution is make them public for use

Comment (jtmoon79) [reply src/naive/datetime/mod.rs:227]: `NANOS_PER_SEC` will be `pub` after #1351

Comment (djc) [reply src/naive/datetime/mod.rs:227]: You can make it `pub(crate)` for this change.

Comment (Ali-Mirghasemi) [reply src/naive/datetime/mod.rs:227]: So which one should i use for more compatibility with new commits?
- add `pub(crate)` in `duration.rs`
or
- make them `pub`

Comment (djc) [reply src/naive/datetime/mod.rs:227]: `pub(crate)`, as said in my previous comment.

**Review by djc — APPROVED** [review]

**Comment by djc** [comment]
Thanks!

**Pull request merged**